hcloud.tf

```diff
@@ -43,11 +64,86 @@ resource "hcloud_server" "k3s" {
   ssh_keys    = [ data.hcloud_ssh_key.ssh_key.id ]
   location    = "${var.location}"
 
+  public_net {
+    ipv4_enabled = true
+    ipv6_enabled = false
+  }  
+
+  network {
+    network_id = hcloud_network.internal-network.id
+    ip         = "10.0.1.6"
+  }
+
   firewall_ids = [ hcloud_firewall.k3s-firewall.id ]
 
   user_data = "${data.template_file.cloud_init.rendered}"
+
+  labels = {
+    label_1 = "${var.env_name}"
+  }
 }
 
 output "ipv4" {
   value = hcloud_server.k3s.ipv4_address
 }
+
+# certificate
+resource "hcloud_managed_certificate" "lb_cert" {
+  name         = "lb_cert"
+  domain_names = ["*.${var.base_domain}", "${var.base_domain}"]
+  labels = {
+    label_1 = "${var.env_name}"
+  }
+}
+
+# load balancer
+resource "hcloud_load_balancer" "wob-tsp-staging" {
+  name               = "wob-tsp-staging"
+  load_balancer_type = "lb11"
+  location           = "${var.location}"
+  labels = {
+    label_1 = "${var.env_name}"
+  }  
+}
+
+# attach LB to network
+resource "hcloud_load_balancer_network" "wob-tsp-staging-srvnetwork" {
+  load_balancer_id = hcloud_load_balancer.wob-tsp-staging.id
+  network_id       = hcloud_network.internal-network.id
+  ip               = "10.0.1.5"
+}
+
+# target for lb
+resource "hcloud_load_balancer_target" "load_balancer_target" {
+  type             = "server"
+  load_balancer_id = hcloud_load_balancer.wob-tsp-staging.id
+  server_id        = hcloud_server.k3s.id
+  use_private_ip   = true
+}
+
+resource "hcloud_load_balancer_service" "load_balancer_service" {
+  load_balancer_id = hcloud_load_balancer.wob-tsp-staging.id
+  protocol         = "https"
+
+  http {
+    redirect_http   = true
+    sticky_sessions = true
+    cookie_name     = "${var.env_name}"
+    certificates    = [hcloud_managed_certificate.lb_cert.id]
+  }
+
+  health_check {
+    protocol = "http"
+    port     = 80
+    interval = 10
+    timeout  = 5
+
+    http {
+      domain       = "${var.base_domain}"
+      path         = "/dontexist" # just check if K3s is up and running
+      response     = "404 page not found"
+      tls          = true
```

Comment:
Does this field make any sense, when `protocol` is set to `http`? The documentation say that it enables "TLS certificate checking"...